rust-runtime/aws-smithy-cbor/benches/string.rs

```diff
@@ -0,0 +1,113 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+pub fn str_benchmark(c: &mut Criterion) {
+    // Definite length key `thisIsAKey`.
+    let definite_bytes = [
+        0x6a, 0x74, 0x68, 0x69, 0x73, 0x49, 0x73, 0x41, 0x4b, 0x65, 0x79,
+    ];
+
+    // Indefinite length key `this`, `Is`, `A` and `Key`.
+    let indefinite_bytes = [
+        0x7f, 0x64, 0x74, 0x68, 0x69, 0x73, 0x62, 0x49, 0x73, 0x61, 0x41, 0x63, 0x4b, 0x65, 0x79,
+        0xff,
+    ];
+
+    c.bench_function("definite str_alt", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&definite_bytes);
+            let x = black_box(decoder.str_alt());
+            assert!(matches!(x.unwrap().as_ref(), "thisIsAKey"));
+        })
+    });
+    c.bench_function("indefinite str_alt", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&indefinite_bytes);
+            let x = black_box(decoder.str_alt());
+            assert!(matches!(x.unwrap().as_ref(), "thisIsAKey"));
+        })
+    });
+
+    c.bench_function("definite str()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&definite_bytes);
+            let x = black_box(decoder.str());
+            assert!(matches!(x.unwrap().as_ref(), "thisIsAKey"));
+        })
+    });
+    c.bench_function("indefinite str()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&indefinite_bytes);
+            let x = black_box(decoder.str());
+            assert!(matches!(x.unwrap().as_ref(), "thisIsAKey"));
+        })
+    });
+}
+
+// We have two `string` implementations. One uses `collect` the other
+// uses `String::new` followed by `string::push`.
+pub fn string_benchmark(c: &mut Criterion) {
+    // Definite length key `thisIsAKey`.
+    let definite_bytes = [
+        0x6a, 0x74, 0x68, 0x69, 0x73, 0x49, 0x73, 0x41, 0x4b, 0x65, 0x79,
+    ];
+
+    // Indefinite length key `this`, `Is`, `A` and `Key`.
+    let indefinite_bytes = [
+        0x7f, 0x64, 0x74, 0x68, 0x69, 0x73, 0x62, 0x49, 0x73, 0x61, 0x41, 0x63, 0x4b, 0x65, 0x79,
+        0xff,
+    ];
+
+    c.bench_function("definite string()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&definite_bytes);
+            let _ = black_box(decoder.str());
+        })
+    });
+
+    c.bench_function("indefinite string()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&indefinite_bytes);
+            let _ = black_box(decoder.str());
+        })
+    });
+
+    c.bench_function("definite string_alt()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&definite_bytes);
+            let _ = black_box(decoder.str_alt());
+        })
+    });
+
+    c.bench_function("indefinite string_alt()", |b| {
+        b.iter(|| {
+            let mut decoder = Decoder::new(&indefinite_bytes);
+            let _ = black_box(decoder.str_alt());
+        })
+    });
+}
+
+pub fn blob_benchmark(c: &mut Criterion) {
```

Comment:
I guess we should put this benchmark in a separate filename? Or maybe modify the filename and benchmark name.